src/integration/java/io/pinecone/integration/dataPlane/ListEndpointTest.java

```diff
@@ -0,0 +1,128 @@
+        // Confirm default vector IDs are returned when no namespace is specified
+        ListResponse listResponseNoArgs = indexConnection.list();
+
+        assertEquals(listResponseNoArgs.getVectorsList().size(), defaultVectorIDs.size());
+
+        assertTrue(listResponseNoArgs.getVectorsList().toString().contains(defaultVectorIDs.get(0)));
+        assertTrue(listResponseNoArgs.getVectorsList().toString().contains(defaultVectorIDs.get(1)));
+        assertTrue(listResponseNoArgs.getVectorsList().toString().contains(defaultVectorIDs.get(2)));
+
+        // Confirm all vector IDs from custom namespace are returned when pass customNamespace
+        ListResponse listResponseCustomNamespace = indexConnection.list(customNamespace);
+
+        assertTrue(listResponseCustomNamespace.getVectorsList().toString().contains(customVectorIDs.get(0)));
+        assertTrue(listResponseCustomNamespace.getVectorsList().toString().contains(customVectorIDs.get(1)));
+        assertTrue(listResponseCustomNamespace.getVectorsList().toString().contains(customVectorIDs.get(2)));
+
+        // Confirm all vector IDs from custom namespace are returned, filtered by given prefix
+        ListResponse listResponseCustomNamespaceWithPrefix = indexConnection.list(customNamespace, "cus-prefix-");
+        List<String> filteredCustomVectorIDs = new ArrayList<>();
+        for (String vectorID : customVectorIDs) {
+            if (vectorID.startsWith("cus-prefix-")) {
+                filteredCustomVectorIDs.add(vectorID);
+            }
+        }
+        assertEquals(listResponseCustomNamespaceWithPrefix.getVectorsList().size(), filteredCustomVectorIDs.size());
+
+        assertTrue(listResponseCustomNamespaceWithPrefix.getVectorsList().toString().contains(customVectorIDs.get(2)));
+        assertTrue(listResponseCustomNamespaceWithPrefix.getVectorsList().toString().contains(customVectorIDs.get(3)));
+
+        // Confirm all vector IDs from custom namespace are returned when limit is specified
+        ListResponse listResponseWithLimit = indexConnection.list(customNamespace, 1);
+
+        assertEquals(1, listResponseWithLimit.getVectorsList().size());
+    }
+
+    @Test
+    public void testAsyncListEndpoint() throws InterruptedException {
+        // Confirm default vector IDs are returned when no namespace is specified
+        ListenableFuture<ListResponse> futureResponseNoArgs = asyncIndexConnection.list();
+        ListResponse asyncListResponseNoArgs = Futures.getUnchecked(futureResponseNoArgs);
+
+        assertEquals(asyncListResponseNoArgs.getVectorsList().size(), defaultVectorIDs.size());
+
+        assertTrue(asyncListResponseNoArgs.getVectorsList().toString().contains(defaultVectorIDs.get(0)));
+        assertTrue(asyncListResponseNoArgs.getVectorsList().toString().contains(defaultVectorIDs.get(1)));
+        assertTrue(asyncListResponseNoArgs.getVectorsList().toString().contains(defaultVectorIDs.get(2)));
```

Comment:
I get why you're using `defaultVectorIDs.get(0)`, etc. everywhere, but I think it makes the test harder to read since it obscures the expectation

Reply:
ah okay! I can hardcode :)